oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -81,4 +102,133 @@ public void multiValuedBinarySize() throws Exception {
+
+        DocumentPropertyState.setCompressionThreshold(DEFAULT_COMPRESSION_THRESHOLD);
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", "\"" + STRING_HUGEVALUE + "\"", mockCompression);
+
+        assertEquals(documentPropertyState.getValue(Type.STRING), STRING_HUGEVALUE);
+
+        verify(mockCompression, times(1)).getOutputStream(any(OutputStream.class));
+    }
+
+    @Test
+    public void uncompressValueThrowsException() throws IOException, NoSuchFieldException, IllegalAccessException {
+
+        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
+        Compression mockCompression = mock(Compression.class);
+        OutputStream mockOutputStream= mock(OutputStream.class);
+        when(mockCompression.getOutputStream(any(OutputStream.class))).thenReturn(mockOutputStream);
+        when(mockCompression.getInputStream(any(InputStream.class))).thenThrow(new IOException("Compression failed"));
+
+        DocumentPropertyState.setCompressionThreshold(DEFAULT_COMPRESSION_THRESHOLD);
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", STRING_HUGEVALUE, mockCompression);
+
+        assertEquals(documentPropertyState.getValue(Type.STRING), "{}");
+
+        verify(mockCompression, times(1)).getInputStream(any(InputStream.class));
+    }
+
+    @Test
+    public void defaultValueSetToMinusOne() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+
+        DocumentPropertyState.setCompressionThreshold(-1);
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyNameNew", "\"" + STRING_HUGEVALUE + "\"", Compression.GZIP);
+
+        byte[] result = state.getCompressedValue();
+
+        assertNull(result);
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+    }
+
+    @Test
+    public void stringAboveThresholdSizeNoCompression() {
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+
+        DocumentPropertyState.setCompressionThreshold(DEFAULT_COMPRESSION_THRESHOLD);
+
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", "\"" + STRING_HUGEVALUE + "\"", Compression.NONE);
+
+        byte[] result = state.getCompressedValue();
+
+        assertEquals(result.length, STRING_HUGEVALUE.length() + 2 );
+
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+        assertEquals(STRING_HUGEVALUE, state.getValue(Type.STRING));
+    }
+
+    @Test
+    public void testInterestingStrings() {
+
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+        String testString = "\"\"simple:foo\", \"cr:a\\n\\b\", \"dquote:a\\\"b\", \"bs:a\\\\b\", \"euro:a\\u201c\", \"gclef:\\uD834\\uDD1E\",\n" +
+                "                \"tab:a\\tb\", \"nul:a\\u0000b\", \"brokensurrogate:\\ud800\"";
```

Comment:
Was it the intention to convert the [original string array](https://github.com/apache/jackrabbit-oak/blob/0e327a27eb988a0f7656535275f3735ef8474e5d/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/BasicDocumentStoreTest.java#L754-L755) into 1 string with escaped double quotes? Shouldn't it be something like below - plus then iterated over each case? 

```suggestion
        String[] tests = new String[] { "simple:foo", "cr:a\n\b", "dquote:a\"b", "bs:a\\b", "euro:a\u201c", "gclef:\uD834\uDD1E",
                "tab:a\tb", "nul:a\u0000b", "brokensurrogate:\ud800" };
```

Reply:
(in which case it will fail)

Reply:
No. The string can have interesting characters in it(it can be string, not string array).

Reply:
sure, it can, but if you test with the original strings the test will fail

Reply:
The test didn't failed for the big string, it compress and uncompress properly.

Reply:
Again, I'd suggest you try the string array as per the original - and try each one individually

Reply:
Done.

Reply:
Maybe we need to clarify first whether we want to keep supporting broken surrogates (for RDBDocumentStore). Before this PR they are supported. With this PR they will no longer be supported, if that particular property value goes through compression/decompression. I don't think the way `isValidSurrogatePair` is currently introduce in the test is correct.

Reply:
Back then (2018) my proposal was to reject string values that do not have an UTF-8 representation, but met resistance from the segment team.

This question affects both segment store and document store, and it would be cool if we could come to consensus about this.

WRT to *this* PR we should just ensure that compression does not affect how MongoDS currently behaves.

Reply:
I support you proposal to reject string values that do not have an UTF-8 representation.